database.go

```diff
@@ -285,39 +325,31 @@ func NewRetentionPolicy(name string) *RetentionPolicy {
 	return &RetentionPolicy{
 		Name:     name,
 		ReplicaN: DefaultReplicaN,
-		SplitN:   DefaultSplitN,
 		Duration: DefaultShardRetention,
 	}
 }
 
-// shardByTimestamp returns the shard in the space that owns a given timestamp for a given series id.
-// Returns nil if the shard does not exist.
-func (rp *RetentionPolicy) shardByTimestamp(seriesID uint32, timestamp time.Time) *Shard {
-	shards := rp.shardsByTimestamp(timestamp)
-	if len(shards) > 0 {
-		return shards[int(seriesID)%len(shards)]
-	}
-	return nil
-}
-
-func (rp *RetentionPolicy) shardsByTimestamp(timestamp time.Time) []*Shard {
-	shards := make([]*Shard, 0, rp.SplitN)
-	for _, s := range rp.Shards {
-		if timeBetweenInclusive(timestamp, s.StartTime, s.EndTime) {
-			shards = append(shards, s)
+// shardGroupByTimestamp returns the shard group in the space that owns a given timestamp for a given series id.
```

Comment:
Also, the reference to "series id" doesn't seem correct -- there is no reference to series ID in this code.